hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

```diff
@@ -895,6 +895,8 @@ void waitForAckedSeqno(long seqno) throws IOException {
     try (TraceScope ignored = dfsClient.getTracer().
         newScope("waitForAckedSeqno")) {
       LOG.debug("{} waiting for ack for: {}", this, seqno);
+      int dnodes = nodes != null ? nodes.length : 3;
+      int writeTimeout = dfsClient.getDatanodeWriteTimeout(dnodes);
```

Comment:
This timeout is very long. For a 3 node pipeline, it will be 8 minutes + 3 * 5 seconds (for the extension).

I'm not sure I have a better suggestion for the timeout.

One question - I believe we saw this problem in a Hung Hive Server 2 process. Do we know how this problem causes the entire HS2 instance to get hung? I would have thought this issue would block the closing of a single file on HDFS and other files open within the same client could still progress as normal?

Reply:
Not much troubleshooting in Hive @sodonnel.  It looks whole HS2 instance was hung. It didn't accept any new connections.

Reply:
Thanks - I guess we can go ahead with this change, but even with it, the HS2 may well get hung for 8+ minutes. Its hard to know for sure without knowing why this problem caused the whole instance to hang.